File: frontend/packages/types/src/admin-cockpit.ts
/** Admin cockpit summary (ADMIN-01A / ADMIN-01B). */

export interface AdminCockpitExecutive {
  users_total: number;
  users_active: number;
  passports_total: number;
  partners_total: number;
  offers_total: number;
  events_total: number;
  creator_contents_total: number;
  partner_leads_total: number;
}

export interface AdminCockpitAttention {
  offers_pending: number;
  creator_contents_pending: number;
  events_pending: number;
  partner_leads_open: number;
  organizations_pending_review: number;
}

export interface AdminCockpitPartners {
  active: number;
  signed: number;
  premium: number;
  founding_partner: number;
  paused: number;
  public: number;
  private: number;
  verified: number;
  pending_review: number;
}

export interface AdminCockpitPassport {
  passports_total: number;
  stamps_total: number;
  qr_stamps: number;
  partner_stamps: number;
  redemptions_total: number;
  redemptions_completed: number;
}

export interface AdminCockpitSummaryResponse {
  generated_at: string;
  city: string;
  executive: AdminCockpitExecutive;
  attention: AdminCockpitAttention;
  partners: AdminCockpitPartners;
  passport: AdminCockpitPassport;
}

export interface AdminCockpitSummaryParams {
  city?: string;
}

File: backend/app/schemas/admin_cockpit.py
"""Admin cockpit summary schemas (ADMIN-01A)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

DEFAULT_COCKPIT_CITY = "Reims"


class AdminCockpitExecutiveMetrics(BaseModel):
    users_total: int = Field(ge=0)
    users_active: int = Field(ge=0)
    passports_total: int = Field(ge=0)
    partners_total: int = Field(ge=0)
    offers_total: int = Field(ge=0)
    events_total: int = Field(ge=0)
    creator_contents_total: int = Field(ge=0)
    partner_leads_total: int = Field(ge=0)


class AdminCockpitAttentionMetrics(BaseModel):
    offers_pending: int = Field(ge=0)
    creator_contents_pending: int = Field(ge=0)
    events_pending: int = Field(ge=0)
    reports_pending: int = Field(ge=0)
    partner_leads_open: int = Field(ge=0)
    organizations_pending_review: int = Field(ge=0)


class AdminCockpitPartnersMetrics(BaseModel):
    active: int = Field(ge=0)
    signed: int = Field(ge=0)
    premium: int = Field(ge=0)
    founding_partner: int = Field(ge=0)
    paused: int = Field(ge=0)
    public: int = Field(ge=0)
    private: int = Field(ge=0)
    verified: int = Field(ge=0)
    pending_review: int = Field(ge=0)


class AdminCockpitPassportMetrics(BaseModel):
    passports_total: int = Field(ge=0)
    stamps_total: int = Field(ge=0)
    qr_stamps: int = Field(ge=0)
    partner_stamps: int = Field(ge=0)
    redemptions_total: int = Field(ge=0)
    redemptions_completed: int = Field(ge=0)


class AdminCockpitTopStampPartner(BaseModel):
    organization_id: str | None = None
    name: str | None = None
    stamps_count: int = Field(ge=0)


class AdminCockpitSignalsMetrics(BaseModel):
    """Derived cockpit signals — real aggregates for dashboard hints (ADMIN-01B+)."""

    offers_published: int = Field(ge=0)
    stamps_today: int = Field(ge=0)
    redemptions_today: int = Field(ge=0)
    passports_last_7_days: int = Field(ge=0)
    events_upcoming: int = Field(ge=0)
    top_stamp_partner: AdminCockpitTopStampPartner


class AdminCockpitSummaryResponse(BaseModel):
    generated_at: datetime
    city: str
    executive: AdminCockpitExecutiveMetrics
    attention: AdminCockpitAttentionMetrics
    partners: AdminCockpitPartnersMetrics
    passport: AdminCockpitPassportMetrics
    signals: AdminCockpitSignalsMetrics
